Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/Analytics.kt

```diff
@@ -67,8 +68,13 @@ class Analytics @Inject constructor(
         }
     }
 
-    suspend fun collectContributions(ppaDataBuilder: PpaData.PPADataAndroid.Builder): List<DonorModule.Contribution> {
-        val request: DonorModule.Request = object : DonorModule.Request {}
+    suspend fun collectContributions(
+        configData: ConfigData,
+        ppaDataBuilder: PpaData.PPADataAndroid.Builder
+    ): List<DonorModule.Contribution> {
+        val request: DonorModule.Request = object : DonorModule.Request {
+            override val currentConfig: ConfigData = configData
+        }
 
```

Comment:
With SAM fun in `Request` we can avoid this boilerplate code, this would simply it to:
```kotlin
       val contributions = donorModules.map {
            Timber.d("Beginning donation for module: %s", it::class.simpleName)
            it.beginDonation { configData }
       }
```

by doing:
```kotlin 
    fun interface Request {
        fun currentConfig(): ConfigData
    }
```

Reply:
Analytics is still a work in progress and we may have to extend the `Request` interface. `SAM fun` only allows a single function, right?

Reply:
Yes, if you expect to add more properties, it is better not to have it.